.github/workflows/static_checks.yml

```diff
@@ -92,3 +92,18 @@ jobs:
       - name: Bazel lint check
         run: |
           bash /home/runner/work/oppia-android/oppia-android/scripts/buildifier_lint_check.sh $HOME
+
+  script_checks:
+    name: Script Checks
+    runs-on: ubuntu-18.04
+    steps:
+      - uses: actions/checkout@v2
+
+      - name: Set up Bazel
+        uses: abhinavsingh/setup-bazel@v3
+        with:
+          version: 4.0.0
+
+      - name: Regex Patterns Validation Check
+        run: |
+          bazel run //scripts:pattern_validation_check -- $(pwd) app data utility domain
```

Comment:
Another thought on usage. Why not just pass in the root directory and perform a recursive check vs. each package?

Reply:
Ah! I thought this too, but there were some good reasons to not go by the recursive checking. First, it saves us from traversing many unwanted directories like `.github`, `.gitsecret`, and many other files at the root level files. So, rather than excluding so many files and directories, it is better to specify only the directories on which we want the particular script check to run. This was also the reason for which the `RepoFile` class was created so that every script check can work only on the directories we specify and not on each and every directory and file. This was the reason why a helper function `checkAllowedDirectory` was introduced. 
This maintains uniformity across all the scripts written as it gives more genericness. So checking every package might be true for the regex check, but it may not be required for the other checks, so if we do this we will be able to maintain uniformity and cleanliness.

Reply:
Hmm, but one big problem with an include-list approach (the current) vs. an exclude-list approach (the one you described) is that in the former we need to remember to update the command to include new directories or they won't be checked. This isn't ideal as it requires a human to remember to do something whereas there's a path forward where we can ensure it's automated to reduce the risk of checks being missed.

To that end, I think the include-all-and-exclude approach is probably more resilient to failure. I definitely agree with your genericness argument, and there may be a way to get both: now that we have a general-purpose repository file class, we can consolidate exclusion to only the directories which should always be ignored by all checks. I think that's .github, .idea, and .bazel* files.

WDYT?

Reply:
Makes sense.
Also, one thing, it might be better not to keep the `.github` directory in the excluded directory, as we might have to recurse over it for the TODO checks, and it possesses no harm to the other script checks.

Reply:
Done.

Reply:
When would we look through ``.github/``? I'm concerned with including that since it includes copies of files from the repository (though not in an easy-to-read format). Given it's mainly Git metadata, it doesn't seem pertinent to static analysis checks or am I missing something?

Reply:
I think this can be resolved now, as per our discussion.